lib/i18n/js/utils.rb

```diff
@@ -1,10 +1,16 @@
 module I18n
   module JS
     module Utils
+      PLURAL_KEYS = %i[zero one two few many other]
+
       # deep_merge by Stefan Rusterholz, see <http://www.ruby-forum.com/topic/142809>.
       # The last result is modified to treat `nil` as missing key
       MERGER = proc do |_key, v1, v2|
-        Hash === v1 && Hash === v2 ? v1.merge(v2, &MERGER) : (v2.nil? ? v1 : v2)
+        if Hash === v1 && Hash === v2
+          (v2.keys - PLURAL_KEYS).empty? ? v2 : v1.merge(v2, &MERGER)
```

Comment:
Please add some code comment to explain the code here
Also please add test case